CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/scatter3d/ArrangeInScatter3dAction.java

```diff
@@ -0,0 +1,267 @@
+import static au.gov.asd.tac.constellation.plugins.arrangements.scatter3d.ArrangeInScatter3dGeneralPlugin.SCATTER_3D_Y_LOGARITHMIC;
+import static au.gov.asd.tac.constellation.plugins.arrangements.scatter3d.ArrangeInScatter3dGeneralPlugin.SCATTER_3D_Z_ATTRIBUTE;
+import static au.gov.asd.tac.constellation.plugins.arrangements.scatter3d.ArrangeInScatter3dGeneralPlugin.SCATTER_3D_Z_LOGARITHMIC;
+import au.gov.asd.tac.constellation.plugins.gui.PluginParametersSwingDialog;
+import au.gov.asd.tac.constellation.plugins.parameters.PluginParameter;
+import au.gov.asd.tac.constellation.plugins.parameters.PluginParameters;
+import au.gov.asd.tac.constellation.plugins.parameters.types.BooleanParameterType;
+import au.gov.asd.tac.constellation.plugins.parameters.types.BooleanParameterType.BooleanParameterValue;
+import au.gov.asd.tac.constellation.plugins.parameters.types.SingleChoiceParameterType;
+import au.gov.asd.tac.constellation.plugins.parameters.types.SingleChoiceParameterType.SingleChoiceParameterValue;
+import java.awt.BorderLayout;
+import java.awt.Component;
+import java.awt.event.ActionEvent;
+import java.awt.event.KeyEvent;
+import java.awt.event.MouseEvent;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.swing.AbstractAction;
+import javax.swing.JButton;
+import javax.swing.JMenu;
+import javax.swing.JMenuBar;
+import javax.swing.JPanel;
+import javax.swing.MenuElement;
+import javax.swing.MenuSelectionManager;
+import org.openide.awt.ActionID;
+import org.openide.awt.ActionReference;
+import org.openide.awt.ActionReferences;
+import org.openide.awt.ActionRegistration;
+import org.openide.util.ImageUtilities;
+import org.openide.util.NbBundle.Messages;
+import org.openide.util.actions.Presenter;
+
+/**
+ * Arranging vertexes in a scatter3d
+ *
+ * @author CrucisGamma
+ */
+@ActionID(category = "Arrange", id = "au.gov.asd.tac.constellation.plugins.arrangements.scatter3d.ArrangeInScatter3dAction")
+@ActionRegistration(displayName = "#CTL_ArrangeInScatter3dAction", iconBase = "au/gov/asd/tac/constellation/plugins/arrangements/scatter3d/resources/scatter3d.png", surviveFocusChange = true)
+@ActionReferences({
+    @ActionReference(path = "Menu/Arrange", position = 0),
+    @ActionReference(path = "Toolbars/Arrange", position = 0),
+    @ActionReference(path = "Shortcuts", name = "S-3")
+})
+@Messages("CTL_ArrangeInScatter3dAction=Scatter3d")
+
+public final class ArrangeInScatter3dAction extends AbstractAction implements Presenter.Toolbar, GraphManagerListener {
+
+    private static final String SCATTER_3D_ACTIONS_ICON = "au/gov/asd/tac/constellation/plugins/arrangements/scatter3d/resources/scatter3d.png";
+    private static final String ATTRIBUTES_SELECTED = "Select Attributes";
+
+    private JPanel panel = null;
+    private JMenuBar menuBar = null;
+    private JMenu menu = null;
+    private Graph graph = null;
+    private JButton chooseDimensions = null;
+
+    public ArrangeInScatter3dAction() {
```

Comment:
This looks more complex than it needs to be - in the simplest case you should be able to mirror what other arrangement plugins are doing (most of them are under 20 lines), and you should be able to auto-generate a gui from the parameters.

Reply:
Have modified this pretty significantly now.